src/main/java/pl/tele/backend/SingleCorrection.java

```diff
@@ -0,0 +1,57 @@
+package pl.tele.backend;
+import java.util.Arrays;
+
+public class SingleCorrection extends Correction{
+    private int[][] hMatrix = {
+            {1, 0, 1, 0, 1, 1, 1, 0, 1, 0, 0, 0},
+            {1, 0, 0, 1, 1, 1, 0, 1, 0, 1, 0, 0},
+            {0, 1, 1, 0, 1, 0, 1, 1, 0, 0, 1, 0},
+            {0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 0, 1},
+    };
+
+    public SingleCorrection() {
+        super(4);
+        super.hMatrix = hMatrix;
+    }
+
+    public String decode(String bitsString) {
+        StringBuilder he = new StringBuilder();
+        for (int i = 0; i < 4; i++) {
+            int rowResult = 0;
+            for (int j = 0; j < 12; j++) {
+                int codedBit = Integer.parseInt(bitsString.substring(j, j + 1));
```

Comment:
```suggestion
                int codedBit = Integer.parseInt(bitsString.charAt(j));
```

Reply:
requires additional change from char to string